pkg/util/future/future_test: TestFuture failed

pkg/util/future/future_test.TestFuture [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9146446?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9146446?buildTab=artifacts#/) on master @ [6839729bc949ad8197319df2371111e8918d89de](https://github.com/cockroachdb/cockroach/commits/6839729bc949ad8197319df2371111e8918d89de):


```
=== RUN   TestFuture
    future_test.go:74: 
        	Error Trace:	pkg/util/future/future_test_test/pkg/util/future/future_test.go:74
        	Error:      	Received unexpected error:
        	            	channel not ready
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | pkg/util/future/future_test_test.TestFuture.func4
        	            	  | 	pkg/util/future/future_test_test/pkg/util/future/future_test.go:67
        	            	  | pkg/util/future/future_test_test.TestFuture
        	            	  | 	pkg/util/future/future_test_test/pkg/util/future/future_test.go:74
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1446
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1594
        	            	Wraps: (2) channel not ready
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
        	Test:       	TestFuture
--- FAIL: TestFuture (0.00s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/unowned
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFuture.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-25664

Epic CRDB-11732

pkg/util/future/future_test.TestFuture [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9250609?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9250609?buildTab=artifacts#/) on master @ [0fcc33bc2870961b9387999d8a9fed97fccbb2ae](https://github.com/cockroachdb/cockroach/commits/0fcc33bc2870961b9387999d8a9fed97fccbb2ae):


```
=== RUN   TestFuture
    future_test.go:75: 
        	Error Trace:	pkg/util/future/future_test_test/pkg/util/future/future_test.go:75
        	Error:      	Received unexpected error:
        	            	channel not ready
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | pkg/util/future/future_test_test.TestFuture.func4
        	            	  | 	pkg/util/future/future_test_test/pkg/util/future/future_test.go:67
        	            	  | pkg/util/future/future_test_test.TestFuture
        	            	  | 	pkg/util/future/future_test_test/pkg/util/future/future_test.go:75
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1446
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1594
        	            	Wraps: (2) channel not ready
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
        	Test:       	TestFuture
--- FAIL: TestFuture (0.00s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFuture.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/util/future/BUILD.bazel
+++ b/pkg/util/future/BUILD.bazel
@@ -20,6 +20,7 @@ go_test(
     deps = [
         ":future",
         "//pkg/testutils",
+        "//pkg/util",
         "//pkg/util/ctxgroup",
         "//pkg/util/leaktest",
         "@com_github_cockroachdb_errors//:errors",

--- a/pkg/util/future/future_test.go
+++ b/pkg/util/future/future_test.go
@@ -20,6 +20,7 @@ import (
 	"time"
 
 	"github.com/cockroachdb/cockroach/pkg/testutils"
+	"github.com/cockroachdb/cockroach/pkg/util"
 	"github.com/cockroachdb/cockroach/pkg/util/ctxgroup"
 	"github.com/cockroachdb/cockroach/pkg/util/future"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
@@ -60,10 +61,14 @@ func TestFuture(t *testing.T) {
 	}()
 
 	mustBeReady := func(ch chan struct{}) error {
+		soon := 5 * time.Second
+		if util.RaceEnabled {
+			soon *= 5
+		}
 		select {
 		case <-ch:
 			return nil
-		default:
+		case <-time.After(soon):
 			return errors.New("channel not ready")
 		}
 	}

